Posts/2021/11/tfl-formatenumeration.md

```diff
@@ -0,0 +1,150 @@
+---
+post_title: How to Use $FormatEnumerationLimit
+username: tfl@psp.co.uk
+Categories: PowerShell
+tags: PowerShell, Format, FormatEnumerationLimit variagble
+Summary: Using The $FormatEnumerationLimit variable in PowerShell
+---
+
+**Q:** When I format an object where a property contains more than 4 objects, I never see the extra property values. How can I fix that?
+
+**A:** Use the `$FormatEnumerationLimit` variable.
+
+This query is one I hear in many PowerShell support forums, and I have encountered this issue a lot over the years.
+What happens is that you issue a command to return objects.
+PowerShell returns the requested objects with properties that are arrays of objects, not a single property value.
+When you pipe those objects to `Format-Table`, by default, PowerShell only shows you the first four
+
+Let me illustrate what this looks like (by default):
+
+```powershell-console
+PS> Get-Process -Name pwsh | Format-Table -Property ProcessName, Modules  
+
+ProcessName Modules  
+----------- -------  
+pwsh        {System.Diagnostics.ProcessModule (pwsh.exe), 
+             System.Diagnostics.ProcessModule (ntdll.dll),
+             System.Diagnostics.ProcessModule (KERNEL32.DLL), 
+             System.Diagnostics.ProcessModule (KERNELBASE.dll)…}
+```
+
+This output shows PowerShell getting the process object for `Pwsh.exe` and then passing it to `Format-Table`, which outputs the process name and four of the modules used by that process.
+However, as you can see, PowerShell shows only four modules shown followed by "…" (also known as an ellipsis).
+The ellipsis tells you that there are more occurrences in this property, except PowerShell does not show them.
+
+If you know the `Format-Table` command, you might be tempted to use the `-Wrap` or the `-AutoSize` parameters, but these would not help.
+It turns out there is no parameter to `Format-Table` or `Format-List` to achieve this ([yet](https://github.com/PowerShell/PowerShell/issues/16360))
```

Comment:
A couple of comments about linking to the issue
1. The link text needs to be more descriptive than "yet" to be accessible. People with low vision that use screen readers will not have the context for that link.
2. Putting the link here disrupts the flow of your narrative. If you want to link to the issue. Talk about it at the end of the post and provide the proper context.

```suggestion
It turns out there is no parameter for `Format-Table` or `Format-List` to control this.
```